msal-dotnet-articles/advanced/high-availability.md

```diff
@@ -95,35 +99,21 @@ Whenever you make **requests for the same token**, i.e. whenever MSAL is able to
 
 Certs for the confidential client app must be rotated for security reasons (don't use secrets in prod!). There are several ways to handle cert rotation, listing is in ordered of most preferred to least preferred.
 
-1. Use Microsoft.Identity.Web's certificate handling logic
+1. Use Managed Identity
+
+With [Managed Identity](https://learn.microsoft.com/entra/msal/dotnet/advanced/managed-identity), trust is established through hosting your app in Azure. There are not secrets to maintain and no certificates to rotate. 
+
+2. Use Microsoft.Identity.Web's certificate handling logic
 
 In web app / web api scenarios, you should use Microsoft.Identity.Web, a higher-level API over MSAL. It handles certificate rotation for when the certificate is stored in KeyVault and handles Managed Identity for you as well.
 
 Learn more in the [Certificates in Microsoft.Identity.Web](https://github.com/AzureAD/microsoft-identity-web/wiki/Certificates#getting-certificates-from-key-vault) guide.
 
 This is the preferred solution for non-Microsoft internal services using ASP.NET Core.
 
-2. (**Internal only**) Rely on [Subject Name/Issuer certificates](./subject-name-and-issuer-authentication.md).
-
-This mechanism allows Azure AD to identify a cert based on SN/I instead of x5t. It is a stop-gap solution, there are no plans to make it available to third-parties.
-
-This is the preferred solution for Microsoft internal services.
-
-3. Write your own simple cert reload logic
-
-Azure AD will reject an expired certificate with an error, which MSAL will report as an `MsalServiceException`
-
-```csharp
-        private bool IsInvalidClientCertificateError(MsalServiceException exMsal)
-        {
-            return !_retryClientCertificate &&
-                string.Equals(exMsal.ErrorCode, "invalid_client", StringComparison.OrdinalIgnoreCase) &&
-                exMsal.Message.Contains("AADSTS700027", StringComparison.OrdinalIgnoreCase);
-        }
-```
+3. (**Internal only**) Rely on [Subject Name/Issuer certificates](./subject-name-and-issuer-authentication.md).
 
-At this stage your app should try to reload the KeyVault certificate.
+This mechanism allows Azure AD to identify a cert based on SN/I instead of thumbprint (x5t). It is a stop-gap solution, there are no plans to make it available to third-parties. 
```

Comment:
```suggestion
This mechanism allows Azure AD to identify a certtificate based on SN/I instead of a thumbprint (x5t). It is a stop-gap solution; there are no plans to make it available to non-Microsoft applications. 
```